### ISSUE TITLE
Refresh our codecov CI

### ISSUE THREAD

**Issue by zbraniecki** [unicode-org/icu4x#2701]
Our codecov CI uses `grcov` with setup from before Rust started supporting `instrument-coverage` natively.

I'm going to check if we can get better coverage if we update to either use modern `grcov` or maybe use an alternative like https://github.com/taiki-e/cargo-llvm-cov

**Comment by sffc** [comment]
According to @ambiguousname we also need to update from v3 to v4 of the GitHub Action.

**Comment by ambiguousname** [comment]
To clarify, the part of the workflow that needs upgrading is codecov/codecov-action@v3. Per https://github.com/marketplace/actions/codecov, a token is required for code coverage. That requires following this guide here, for whoever has access to the repo's secrets: https://docs.codecov.com/docs/github-2-getting-a-codecov-account-and-uploading-coverage

**Comment by ambiguousname** [comment]
Although it seems like there's some redundancy in the workflow, since grcov seems to be able to push to both coveralls and codecov, so we could just remove the part of the workflow that pushes to codecov.

**Comment by sffc** [comment]
Good point! I changed https://github.com/unicode-org/icu4x/pull/4922 to keep uploading to coveralls and just not codecov. I think we had it uploading to both tools because we couldn't decide which one was better.
